lib/Sema/TypeCheckDeclOverride.cpp

```diff
@@ -1672,6 +1672,20 @@ static bool checkSingleOverride(ValueDecl *override, ValueDecl *base) {
     }
   }
 
+  if (auto VD = dyn_cast<VarDecl>(override)) {
+    auto baseVD = cast<VarDecl>(base);
+    if (auto overrideReadCoroutine = VD->getReadCoroutine()) {
+      if (auto baseReadCoroutine = baseVD->getReadCoroutine()) {
+        if (overrideReadCoroutine->hasThrows() &&
+            !baseReadCoroutine->hasThrows()) {
+          diags.diagnose(overrideReadCoroutine, diag::override_throws,
+                         isa<ConstructorDecl>(override));
+          diags.diagnose(baseReadCoroutine, diag::overridden_here);
+        }
+      }
+    }
+  }
```

Comment:
Why is this specific to read coroutines?

Reply:
@rjmccall The check is already handled for `_modify`, but I haven't figured out why it doesn't work for `_read` so I had to add this check.